### DISCUSSION TITLE
trigger method in parent from child nested in router in Quasar (vue)

### DISCUSSION THREAD

**Discussion by perottosson** [quasarframework/quasar#9649] [General - Components / Directives / etc]
I got this structure with nested routes in my router using Quasar framework (vue 3):

```
    const routes = [
      {
        path: "/",
        component: () => import("layouts/myLayout.vue"),
        children: [
          {
            path: "",
            component: () => import("pages/Main.vue"),
            children: [
              {
                path: "",
                component: () => import("components/Sub.vue")
              }
            ]
          }
        ]
      }
```

I know I can use @emit in my child to pass to parent like this:

**MyChild:**

`      this.$emit("myEvent", "hello world");
`
**MyParent:**

`    <MyChild @myEvent="updateMyEvent" />
`
However I want to trigger an event in parent without rendering MyChild once again in the parent since it is already displayed through the router ... So I am looking for a way of accessing the parents method more directly so to speak. 

What is the correct implementation of doing this in vue 3?

**Comment by hawkeye64** [comment]
@perottosson Look into the provide/inject methods
